tests/sycl/reduction.cpp

```diff
@@ -327,6 +328,12 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(two_reductions, T, large_test_types) {
   test_two_reductions<T>(128*128, 128);
 }
 
+#ifdef BOOST_SIGACTION_BASED_SIGNAL_HANDLING
+BOOST_AUTO_TEST_CASE(local_size_one_reduction, * boost::unit_test::timeout(2)) {
```

Comment:
I have added a test case that has a timeout of 2 seconds to detect if the reduction is stuck in an infinite loop.

The `#ifdef' checks if boost is actually able to force stop the test if it times out, otherwise the test would be reported as failed but the test suite would still continue execution (in the infinite loop). I don't know however when `BOOST_SIGACTION_BASED_SIGNAL_HANDLING` is actually defined, on my setup on jarvis, for instance, it's not. So I'm not sure if this test makes sense this way.